recipes/ios-cmake/all/conanfile.py

```diff
@@ -0,0 +1,94 @@
+    default_options = {
+        "enable_bitcode": True,
+        "enable_arc": True,
+        "enable_visibility": False,
+        "enable_strict_try_compile": False,
+        "ios_target": "OS64COMBINED", 
+    }
+    description = "ios Cmake toolchain to (cross) compile macOS/iOS/watchOS/tvOS"
+    topics = "conan", "apple", "ios", "cmake", "toolchain", "ios", "tvos", "watchos"
+    exports_sources =  "cmake-wrapper"
+
+    @property
+    def _source_subfolder(self):
+        return os.path.join(self.source_folder, "source_subfolder")
+
+    @staticmethod
+    def _chmod_plus_x(filename):
+        if os.name == 'posix':
+            os.chmod(filename, os.stat(filename).st_mode | 0o111)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("ios-cmake-{}".format(self.version), self._source_subfolder)
+
+    def build(self):
+        pass # there is nothign to build
+
+    def package(self):
+        self.copy("cmake-wrapper", dst="bin")
+        self.copy("ios.toolchain.cmake", 
+                    src=self._source_subfolder,  
+                    dst=os.path.join("lib", "cmake", "ios-cmake"), 
+                    keep_path=False)
+        self._chmod_plus_x(os.path.join(self.package_folder, "bin", "cmake-wrapper"))
+
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder, keep_path=False)
+
+
+    def package_info(self):
+        arch_flag = self.settings.arch
+        if arch_flag == "armv8":
+            arch_flag = "arm64"
+
+        cmake_options = "-DENABLE_BITCODE={} -DENABLE_ARC={} -DENABLE_VISIBILITY={} -DENABLE_STRICT_TRY_COMPILE={}".format(
+            self.options.enable_bitcode, 
+            self.options.enable_arc, 
+            self.options.enable_visibility,
+            self.options.enable_strict_try_compile  
+        ) 
+        # Note that this, as long as we specify (overwrite) the ARCHS, PLATFORM has just limited effect, 
+        # but PLATFORM need to be set in the profile so it makes sense, see ios-cmake docs for more info
+        cmake_flags = "-DPLATFORM={} -DDEPLOYMENT_TARGET={} -DARCHS={} {}".format(
+            self.options.ios_target, self.settings.os.version, arch_flag, cmake_options
+        ) 
+
+        self.env_info.CONAN_USER_CMAKE_FLAGS = cmake_flags
+        self.output.info("Setting toolchain options to: {}".format(cmake_flags))
+        cmake_wrapper = os.path.join(self.package_folder, "bin", "cmake-wrapper")
+        self.output.info("Setting CONAN_CMAKE_PROGRAM to: {}".format(cmake_flags))
+        self.env_info.CONAN_CMAKE_PROGRAM = cmake_wrapper
+        tool_chain = os.path.join(self.package_folder,
+                                    "lib", 
+                                    "cmake", 
+                                    "ios-cmake",
+                                    "ios.toolchain.cmake")
+        self.env_info.CONAN_CMAKE_TOOLCHAIN_FILE = tool_chain
+
+    def package_id(self):
+        self.info.header_only()
```

Comment:
You need to zap the header_only()

Reply:
but how to make the package aware that there is just 1 build needed that works for all configurations of iOS (and Watch + TV OS_ ?

Reply:
Because this is OS/arch dependent, I dont think having just one is the right solution

```
cmake_flags = "-DPLATFORM={} -DDEPLOYMENT_TARGET={} -DARCHS={} {}".format(
            self.options.ios_target, self.settings.os.version, arch_flag, cmake_options
        ) 
```

Reply:
@prince-chrismc 
The cmake script needs to know what arch is used.
Whatever os/arch/compiler is used, the cmake script stays the same.
So we want the `header_only` behavior.

Reply:
RIP okay, it seems counter intuitive that it needs to own it but it stays the same :confused: 

Sadly it still leaves us with the hooks not passing

Reply:
A hack would be to create a dummy header in the `include` folder.
Adding something like:
```python
# satisfy KB-H014 (header_only recipes require headers
os.mkdir(os.path.join(self.package_folder, "include"))
tools.save(os.path.join(self.package_folder, "include", "dummy_header.h"))
```

Reply:
also thought about such a hack, and maybe I will add it since it seems waiting that conan chages and accepts the reality that there are tools that work for more archs, but are not header only, might take too long